api.bs

```diff
@@ -142,8 +167,147 @@ The aggregation service:
 
 # API Details # {#api}
 
+Open questions:
+*   Filter/query language
+*   Reports are sent to aggregation system directly, or via conversion site? Or
+    option of either? => via conversion site
+*   Epochs
+
 TODO
 
+## ListAggregationSystems API ## {#list-aggregation-systems-api}
+
+navigator.privateAttribution.listAggregationSystems()
+
+<xmp class=idl>
+dictionary PrivateAttributionAggregationSystem {
+  required DOMString id;
+};
+</xmp>
+
+## SaveImpression API ## {#save-impression-api}
+
+<pre>
+navigator.privateAttribution.saveImpression({
+  type: "view",                  // either "view" or "click"
+  index: 3,                      // the histogram index for counting this impression
+  ad: "sample-campaign-eijb",    // a unique identifier for the ad placement
+  target: "advertiser.example",  // the advertiser site where a conversion will occur
+});
+</pre>
+
+Add:
+* attribution system
+* TTL
+* DP parameters
+
+Questions:
+*   Revisit the set of impression types. Can we get rid of it, and put it in the
+    ad ID? Or generalize to "attribution constraint"?
+
+<xmp class=idl>
+enum PrivateAttributionImpressionType { "view", "click" };
+
+dictionary PrivateAttributionImpressionOptions {
+  PrivateAttributionImpressionType type = "view";
+  required unsigned long index;
+  required DOMString ad;
+  required DOMString target;
+};
+
+[SecureContext, Exposed=Window]
+interface PrivateAttribution {
+  [Throws] undefined saveImpression(DOMString aggregationSystemId, PrivateAttributionImpressionOptions options);
```

Comment:
```suggestion
  [Throws] undefined saveImpression(PrivateAttributionImpressionOptions options);
```

Given that we have a bucket of fields, some of which are mandatory, let's move the aggregation system into the bucket.

Except... We don't need to know which system is being used, do we?  I don't think we care: budgeting happens at the point of conversion, so we probably don't care where conversion data is sent as long as the aggregation system is trustworthy.  But at impression time, we really don't care.  Yet.  In the full API we will.  But the bag of attributes method lets us solve that problem neatly.

Reply:
I guess we don't strictly need it here in the early binding case, but given that we expect to have it eventually, it seems simpler to include it. It seems better to encourage users to register an impression for each aggregator than to set a precedent of supporting the model where the same impression can be referenced by multiple different aggregators at conversion time. There isn't a lot of complexity in the impression data right now, but what happens if the aggregators want to set conditions (like a limit on impression TTL, or the structure of ad IDs)?